extensions/security/runtime/src/main/java/io/quarkus/security/runtime/QuarkusSecurityIdentity.java

```diff
@@ -205,6 +207,57 @@ public Builder addAttributes(Map<String, Object> attributes) {
             return this;
         }
 
+        /**
+         * Adds a permission.
+         *
+         * @param permission The permission
+         * @return This builder
+         */
+        public Builder addPermission(String permission) {
+            return addStringPermissions(Set.of(permission));
+        }
+
+        /**
+         * Adds permissions.
+         *
+         * @param permissions The permissions
+         * @return This builder
+         */
+        public Builder addStringPermissions(Set<String> permissions) {
+            return addPermissions(permissions.stream().map(p -> new StringPermission(p))
+                    .collect(Collectors.toSet()));
+        }
+
+        /**
+         * Adds a permission.
+         *
+         * @param permission The permission
+         * @return This builder
+         */
+        public Builder addPermission(Permission permission) {
+            return addPermissions(Set.of(permission));
+        }
+
+        /**
+         * Adds permissions.
+         *
+         * @param permissions The permissions
+         * @return This builder
+         */
+        public Builder addPermissions(Set<Permission> permissions) {
+            for (Permission permission : permissions) {
+                addPermissionChecker(new Function<Permission, Uni<Boolean>>() {
+
+                    @Override
+                    public Uni<Boolean> apply(Permission requiredPermission) {
+                        return Uni.createFrom().item(permission.implies(requiredPermission));
```

Comment:
Strictly speaking this should be fine because we resolve permission checkers uni in parallel. What I thought would be more efficient, but also more verbose, would be to:

- keep `Set<Permission>` on the `Builder` instance
- when `build()` is called, you create one synchronous permission checker like this:

```
for (Permission possessedPermission : permissions) {
    if (possessedPermission.implies(requiredPermission)) {
        return Uni.createFrom().item(true);
    }
}
return Uni.createFrom().item(false);
```
and you add that checker to other checkers before you build `SecurityIdentity`. It would avoid `Uni`. I don't see cons on such approach, however I think my suggestions have so little pros that you don't need to apply it.

Reply:
Sure, added your proposal @michalvavrik, have a look please

Reply:
looks good, I can't close this comment thread, please consider this thread resolved